src/basecamp.cpp

```diff
@@ -23,24 +68,98 @@ std::string basecamp::board_name() const
     return string_format( _( "%s Board" ), name.c_str() );
 }
 
-std::string basecamp::save_data() const
+#include <stdio.h>
+void basecamp::define_camp( npc &p )
+{
+    const auto parse_expansion = []( const std::string &terrain ) {
+        expansion_data e;
+        int last_bar = terrain.find_last_of( '_' );
+        e.type = terrain.substr( 13, last_bar - 13 );
+        e.cur_level = std::stoi( terrain.substr( last_bar + 1 ) );
+        return e;
+    };
+    sort_points = p.companion_mission_points;
+    // purging the regions guarantees all entries will start with faction_base_
+    for( std::pair<std::string, tripoint> expansion : talk_function::om_building_region( p, 1,
+            true ) ) {
+        const std::string dir = talk_function::om_simple_dir( pos, expansion.second );
+        expansions[ dir ] = parse_expansion( expansion.first );
+        expansion_data &e = expansions[ dir ];
+        printf( "adding %s at %s at %d\n", e.type.c_str(), dir.c_str(), e.cur_level );
```

Comment:
This will mess up the output in ncurses-builds badly.

Reply:
temporary debugging code.